// @bairong/dialog_mask/lib/index.js
var $ = require('jquery')

/**
  * mask遮罩
  */
var mask = {
	crrentHeight : 0,
	defaultStyle : function(){
		return {
			height : "auto",
			zIndex : 1000,
			display: 'block',
			opacity: 0.7,
			background : '#000',
			position :'absolute',
			top : 0,
			right : 0,
			bottom :0,
			left : 0,
		}
	},
	creatMask : function(pn,cg){
		if(!pn) {
			throw new Error('parent element is null ! plase creat element  fill secend param(only id)');
		};
		var wrapName,maskElm; 
		wrapName = $("#" + pn);
		maskElm = document.createElement('div');
		maskElm.id = 'maskDialog';
		wrapName.append(maskElm);
		
		var maskConfig = cg || Object.keys(cg).length || this.defaultStyle();
		this.maskAddStyle(maskConfig);
	},
	addStyle : function(styleList){
		var elm = $("#maskDialog");
		if(!$("#maskDialog").size()){
			throw new Error('the id#maskDialog is no defined');
			return
		}
		elm.css(styleList);
	},
	maskAddStyle : function(maskConfig){
		if(!maskConfig || !Object.keys(maskConfig).length) {
			this.addStyle(this.defaultStyle());
			return;
		};
		var initStyle = this.defaultStyle();
		var styleList = {};
		
		for(var k in initStyle){
			if(!maskConfig[k]){
				styleList[k] = initStyle[k];
			}
			else{
				styleList[k] = maskConfig[k];
			}
		}
		styleList.height = this.isLessClient(styleList.height);
		this.addStyle(styleList);
	},
	isLessClient : function(setHeight){
		var clientHeight = document.documentElement.clientHeight;
		if(setHeight < clientHeight){
			this.crrentHeight = clientHeight;
			return clientHeight;
		}
		this.crrentHeight = setHeight;
		return setHeight;
	},
	getHeight : function(){
		return this.crrentHeight;
	}
}

/**
 * 没有id为maskDialog的元素，先创建一个父元素把名字填入第二个参数,第一个为样式列表Json
 */
const maskCtrl = {
	show : function(config,pushElmName){
		if(!document.getElementById("maskDialog")){
			mask.creatMask(pushElmName,config);
		}
		else{
			mask.maskAddStyle(config);
		}
	},
	hidden : function(){
		var recoverHeight = document.documentElement.clientHeight;
		mask.maskAddStyle({display:"none",height:recoverHeight});
	},
	getHeight : function(){
		return mask.getHeight();
	}
}

module.exports = maskCtrl;
//if(typeof exports === 'object' && typeof module === 'object'){
//	
//}
//else{
//	export default maskCtrl;
//}
